### ISSUE TITLE
TRAFFIC tab crashed when Talkback is enabled.

### ISSUE THREAD

**Issue by irfano** [wordpress-mobile/WordPress-Android#20287]
### Expected behavior
The traffic tab should work with accessibility.

### Actual behavior
When Talkback is enabled, the app crashes on opening the Traffic Tab.

### Steps to reproduce the behavior
1. Log in JP app.
2. Enable "stats_traffic_tab" from "Me → Debug settings"
3. Enable Talkback from the device settings.
4. Open "My Site → Stats"
5. Open the TRAFFIC tab and select By week, By month or By year.
6. Notice it crashes.

https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/65386de6-2518-4e4b-beeb-46c89800afb7

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by aditi-bhatia** [comment]
I tried the steps above on a few different devices (both physical and emulators), including one Pixel 7 Pro emulator running Android 14, and was unable to reproduce on any of them. I tested with the stats on `mobile.blog` and I'm using the latest master. Are you able to reproduce consistently on other devices?

**Comment by irfano** [comment]
I added a video to the descriptio. Here is the crash log:

```
E  FATAL EXCEPTION: main
                 Process: com.jetpack.android.beta, PID: 6917
                 java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1
                 	at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
                 	at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
                 	at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
                 	at java.util.Objects.checkIndex(Objects.java:359)
                 	at java.util.ArrayList$SubList.get(ArrayList.java:1166)
                 	at org.wordpress.android.ui.stats.refresh.utils.BarChartAccessibilityHelper.onPopulateNodeForVirtualView(BarChartAccessibilityHelper.kt:67)
                 	at androidx.customview.widget.ExploreByTouchHelper.createNodeForChild(ExploreByTouchHelper.java:805)
                 	at androidx.customview.widget.ExploreByTouchHelper.obtainAccessibilityNodeInfo(ExploreByTouchHelper.java:725)
                 	at androidx.customview.widget.ExploreByTouchHelper$MyNodeProvider.createAccessibilityNodeInfo(ExploreByTouchHelper.java:1253)
                 	at androidx.core.view.accessibility.AccessibilityNodeProviderCompat$AccessibilityNodeProviderApi16.createAccessibilityNodeInfo(AccessibilityNodeProviderCompat.java:51)
```

When I fix this specific `IndexOutOfBoundsException` exception, it throws another exception. We copied the bar chart on the TRAFFIC tab from the `OverViewUseCase`, it's possible that we may have missed something while copying the chart.

**Comment by aditi-bhatia** [comment]
Hi @irfano I don't see a video in the description 🤔 Can you confirm that you're able to reproduce this consistently on other devices?

**Comment by aditi-bhatia** [comment]
This is a video with what I'm seeing (no errors in log):

https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/52fe37f9-9f1c-46f6-aeea-756a7f82950c

**Comment by irfano** [comment]
That's strange! 🤔 It's a separate issue. The page should update when changing the granularity. Did you record this video from the `trunk` version?

**Comment by irfano** [comment]
> Hi @irfano I don't see a video in the description

It seems the upload failed last time. I have uploaded it now.

**Comment by aditi-bhatia** [comment]
> That's strange! 🤔 It's a separate issue. The page should update when changing the granularity. Did you record this video from the trunk version?

Yes, this is while on the latest trunk. I think it's possibly related to the same issue since the issue only happens with `talkback` enabled

**Comment by irfano** [comment]
> I think it's possibly related to the same issue since the issue only happens with talkback enabled

Yes, it is possible. Can you debug it? `TrafficListViewModel.onGranularitySelected()` is the function responsible for changing granularity.

**Comment by aditi-bhatia** [comment]
@irfano Yup, looking into it 👍
